test/unittest/statistics/rtps/MonitorServiceTests.cpp

```diff
@@ -0,0 +1,274 @@
+                    [this](std::vector<fastrtps::rtps::GUID_t>& guids)
+                    {
+                        guids.reserve(n_local_entities);
+                        mock_guids.reserve(n_local_entities);
+
+                        for (size_t i = 1; i <= static_cast<size_t>(n_local_entities); i++)
+                        {
+                            fastrtps::rtps::GUID_t guid;
+                            guid.entityId.value[3] = i;
+                            guids.push_back(guid);
+                            mock_guids.push_back(guid);
+                        }
+                        return true;
+                    }));
+    }
+
+    void TearDown() override
+    {
+
+    }
+
+protected:
+
+    MockConnectionsQueryable mock_conns_q_;
+    MockStatusQueryable mock_status_q_;
+    MockProxyQueryable mock_proxy_q_;
+
+    MonitorService monitor_srv_;
+    MonitorServiceListener listener_;
+    int n_local_entities;
+    std::vector<fastrtps::rtps::GUID_t> mock_guids;
+};
+
+TEST_F(MonitorServiceTests, enabling_monitor_service_routine)
+{
+    //! At the startup, the service should collect the already existing
+    //! local entities
+    EXPECT_CALL(mock_proxy_q_, get_all_local_proxies(::testing::_)).Times(1);
+    EXPECT_CALL(mock_proxy_q_, get_serialized_proxy(::testing::_, ::testing::_)).Times(n_local_entities);
+    EXPECT_CALL(mock_conns_q_, get_entity_connections(::testing::_)).Times(n_local_entities);
+
+    //! Enable the service
+    ASSERT_FALSE(monitor_srv_.disable_monitor_service());
+    ASSERT_TRUE(monitor_srv_.enable_monitor_service());
+    ASSERT_TRUE(monitor_srv_.is_enabled());
+
+    //! Verify expectations
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+    ::testing::Mock::VerifyAndClearExpectations(&mock_conns_q_);
+    ::testing::Mock::VerifyAndClearExpectations(&mock_proxy_q_);
+
+}
+
+TEST_F(MonitorServiceTests, multiple_proxy_and_connection_updates)
+{
+    //! Enable the service
+    ASSERT_FALSE(monitor_srv_.disable_monitor_service());
+    ASSERT_TRUE(monitor_srv_.enable_monitor_service());
+    ASSERT_TRUE(monitor_srv_.is_enabled());
+
+    //! Skip initial transient
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+
+    //! Expect the getters for each status that is going to be updated
+    EXPECT_CALL(mock_proxy_q_, get_serialized_proxy(::testing::_, ::testing::_)).
+            Times(n_local_entities);
+    EXPECT_CALL(mock_conns_q_, get_entity_connections(::testing::_)).
+            Times(n_local_entities);
+
+    //! Trigger statuses updates for each entity
+    for (auto& entity : mock_guids)
+    {
+        listener_.on_local_entity_change(entity, true);
+        listener_.on_local_entity_connections_change(entity);
+    }
+
+    //! Verify expectations
+    std::this_thread::sleep_for(std::chrono::seconds(5));
+    ::testing::Mock::VerifyAndClearExpectations(&mock_status_q_);
```

Comment:
Verify `mock_proxy_q_` and `mock_conns_q_` as well. Also, same question regarding asserts.

Reply:
In fact, only `mock_proxy_q_` and `mock_conns_q_` are required to be checked here